### ISSUE TITLE
RPR 2.20 and Houdini 596

### ISSUE THREAD

**Issue by wmosx** [GPUOpen-LibrariesAndSDKs/RadeonProRenderUSD#487]
I have never been able to get this to work correctly since 2.027 & Houdini 18.5.499

Whether its just adding a sphere and selecting RPR or using test geometry with RPR material and render settings.

Please advise if I am doing something wrong or my settings are incorrect.

Hardware:
MacPro 6.1 
BigSur 11.4
2.7GHZ 12 Core 
128GB 
Dual D700
AMD WX8200 EGPU recognized as AMD Radeon Vega

Thank you,
William

RPR JSON

{"env":[
{"RPR":"/Users/wmosx/Library/Preferences/houdini/18.5/hdRpr-2.2.0-Houdini-18.5.596-py2.7-macOS"},
{"HOUDINI_PATH":"/Users/wmosx/Library/Preferences/houdini/18.5/hdRpr-2.2.0-Houdini-18.5.596-py2.7-macOS/houdini"},
{"PYTHONPATH":"/Users/wmosx/Library/Preferences/houdini/18.5/hdRpr-2.2.0-Houdini-18.5.596-py2.7-macOS/lib/python"}]}

RPR_Materials JSON
{
    "env":[
        {
            "RPR_MTLX_MATERIAL_LIBRARY_PATH":"/Users/wmosx/Library/Preferences/houdini/18.5/RPR_MaterialLibrary"
        }
    ]
}

Crash report from wmosx; Houdini Version 18.5.596 [macosx10.14-x86_64-clang10.0-targetosx10.12]
Uptime 124 seconds
Tue Jul 13 22:37:28 2021
Caught signal 11

2   libHoudiniUI.dylib                  0x000000010b8752b1 AP_Interface::si_CrashHandler::handledSignal(UTsignalHandlerArg) + 561
3   libHoudiniUT.dylib                  0x000000010e2fbbd4 signalCallback(UTsignalHandlerArg) + 84
4   libHoudiniUT.dylib                  0x000000010e680640 UT_Signal::processSignal(int, __siginfo*, void*) + 192
5   libsystem_platform.dylib            0x00007fff2037ad7d _sigtramp + 29
6   ???                                 0x0000000000000000 0x0 + 0
7   Metal                               0x00007fff28530739 -[MTLIOAccelCommandBuffer didCompleteWithStartTime:endTime:error:] + 104
8   Metal                               0x00007fff28530622 -[_MTLCommandQueue commandBufferDidComplete:startTime:completionTime:error:] + 161
9   IOAccelerator                       0x00007fff28508cd6 ioAccelCommandQueueBlockFenceCallback + 44
10  IOKit                               0x00007fff22a156cb IODispatchCalloutFromCFMessage + 364
11  IOKit                               0x00007fff22a1554c _IODispatchCalloutWithDispatch + 33
12  libdispatch.dylib                   0x00007fff201a42f8 dispatch_mig_server + 350
13  libdispatch.dylib                   0x00007fff2018b806 _dispatch_client_callout + 8
14  libdispatch.dylib                   0x00007fff2018e1b0 _dispatch_continuation_pop + 423
15  libdispatch.dylib                   0x00007fff2019e564 _dispatch_source_invoke + 2061
16  libdispatch.dylib                   0x00007fff20191493 _dispatch_lane_serial_drain + 263
17  libdispatch.dylib                   0x00007fff201920ad _dispatch_lane_invoke + 366
18  libdispatch.dylib                   0x00007fff2019bc0d _dispatch_workloop_worker_thread + 811
19  libsystem_pthread.dylib             0x00007fff2033245d _pthread_wqthread + 314
20  libsystem_pthread.dylib             0x00007fff2033142f start_wqthread + 15

2   libHoudiniUI.dylib                  0x000000010b873521 AP_Interface::coreDumpChaser(UTsignalHandlerArg) + 641
3   libHoudiniUI.dylib                  0x000000010b87535e AP_Interface::si_CrashHandler::chaser(UTsignalHandlerArg) + 46
4   libHoudiniUT.dylib                  0x000000010e2fbd5d signalCallback(UTsignalHandlerArg) + 477
5   libHoudiniUT.dylib                  0x000000010e680640 UT_Signal::processSignal(int, __siginfo*, void*) + 192
6   libsystem_platform.dylib            0x00007fff2037ad7d _sigtramp + 29
7   ???                                 0x0000600000684480 0x0 + 105553123099776
8   CoreFoundation                      0x00007fff2042e8c5 __CFRunLoopServiceMachPort + 316
9   CoreFoundation                      0x00007fff2042cf7b __CFRunLoopRun + 1332
10  CoreFoundation                      0x00007fff2042c380 CFRunLoopRunSpecific + 567
11  HIToolbox                           0x00007fff288ccab3 RunCurrentEventLoopInMode + 292
12  HIToolbox                           0x00007fff288cc815 ReceiveNextEventCommon + 587
13  HIToolbox                           0x00007fff288cc5b3 _BlockUntilNextEventMatchingListInModeWithFilter + 70
14  AppKit                              0x00007fff22b576f2 _DPSNextEvent + 864
15  AppKit                              0x00007fff22b55ec5 -[NSApplication(NSEvent) _nextEventMatchingEventMask:untilDate:inMode:dequeue:] + 1364
16  libqcocoa.dylib                     0x0000000154d3e43f qt_plugin_instance + 182127
17  libHoudiniUI.dylib                  0x000000010b5565b5 UI_QtGeneratorThread::waitForRawEvents() + 373
18  libHoudiniUI.dylib                  0x000000010b663132 UI_Queue::waitForEvents() + 114
19  libHoudiniUI.dylib                  0x000000010b662497 UI_Queue::eventLoop() + 375
20  libHoudiniUI.dylib                  0x000000010b8784d6 main + 5302
21  libdyld.dylib                       0x00007fff20350f5d start + 1

![Screen Shot 2021-07-13 at 10 37 46 PM](https://user-images.githubusercontent.com/84696482/125552159-27dc41aa-faa2-4287-b16a-3d7e7324bc00.png)

**Comment by hshakula** [comment]
I assume that you experience some issues due to multi-gpu interoperability. Currently, we always use all available GPUs in the plugin. Could you please try this [build](https://drive.google.com/file/d/1UncbzCClAFCOunm-oUvQAx8q8Ztab9zU/view?usp=sharing)?

**Comment by hshakula** [comment]
I assume that you experience this issue due to multi-gpu interoperability. Currently, we always use all available GPUs in the plugin. Could you please try this [build](https://drive.google.com/file/d/1UncbzCClAFCOunm-oUvQAx8q8Ztab9zU/view?usp=sharing)?

**Comment by wmosx** [comment]
That is correct I am using a 2013 MacPro with Dual D700’s as well as and External AMD WX8200 EGPU.

Houdini is configured to used the external card.  Will give it a go and report back.

Thank You,
William

On Jul 18, 2021, at 18:20, George Shakula ***@***.***> wrote:

﻿

I assume that you experience some issues due to multi-gpu interoperability. Currently, we always use all available GPUs in the plugin. Could you please try this build<https://drive.google.com/file/d/1UncbzCClAFCOunm-oUvQAx8q8Ztab9zU/view?usp=sharing>?

**Comment by wmosx** [comment]
Confirming issues has been resolved with the provided build.

I was able to render to 100% for the first time since installing the addon as well testing with multiple materials from the RPR Library.

Thank you so much and enjoy the rest of your weekend!

Best Regards,
William
________________________________

**Comment by hshakula** [comment]
Hey, I've added a GUI to choose exact devices that should be used by RPR for rendering (https://github.com/GPUOpen-LibrariesAndSDKs/RadeonProRenderUSD/pull/496).

Would you mind giving it a try? [Preview build](https://drive.google.com/file/d/1YSwEfQa287i63ssYSp2ujBEEF5uSQrL3/view?usp=sharing).

Please note that the build I gave you before was using only the first GPU. So you can start by configuring it the same way.
